rlp/encode_test.go

```diff
@@ -72,12 +72,28 @@ type encTest struct {
 }
 
 var encTests = []encTest{
+	// bytes and byte slices
+	{val: byte(0), output: "80"},
+	{val: byte(0x0), output: "80"},
+	{val: byte(0x00), output: "80"},
+	{val: byte(1), output: "01"},
+	{val: byte(0x1), output: "01"},
+	{val: byte(0x01), output: "01"},
+	{val: byte(0x7f), output: "7F"},
+
+	{val: []byte{}, output: "80"},
+	{val: []byte{0}, output: "00"}, // PTAL
```

Comment:
```suggestion
	{val: []byte{0}, output: "00"},
```
This is perfectly correct, but a bit surprising at first :joy: